src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3001,7 +3002,50 @@ namespace winrt::TerminalApp::implementation
         ShowWindowChanged.raise(*this, args);
     }
 
-    winrt::fire_and_forget TerminalPage::_SearchMissingCommandHandler(const IInspectable /*sender*/, const Microsoft::Terminal::Control::SearchMissingCommandEventArgs args)
+    Windows::Foundation::IAsyncOperation<IVectorView<MatchResult>> TerminalPage::_FindPackagesInCatalogAsync(PackageCatalog catalog, PackageMatchField field, PackageFieldMatchOption matchOption, hstring query)
+    {
+        FindPackagesOptions findPackagesOptions = WindowsPackageManagerFactory::Instance().CreateFindPackagesOptions();
+        PackageMatchFilter filter = WindowsPackageManagerFactory::Instance().CreatePackageMatchFilter();
+        filter.Field(field);
+        filter.Option(matchOption);
+        filter.Value(query);
+        findPackagesOptions.Filters().Append(filter);
+        findPackagesOptions.ResultLimit(20);
+        FindPackagesResult findPackagesResult{ co_await catalog.FindPackagesAsync(findPackagesOptions) };
+
+        co_return findPackagesResult.Matches();
+    }
+
+    Windows::Foundation::IAsyncOperation<IVectorView<MatchResult>> TerminalPage::_FindPackageAsync(hstring query)
+    {
+        co_await winrt::resume_background();
```

Comment:
Considering that `_SearchMissingCommandHandler` already asserts that it's on a background thread in the beginning, this may be superfluous.